website/content/en/docs/olm-integration/generating-a-csv.md

```diff
@@ -1,208 +1,248 @@
+_For Go Operators only:_ Several fields require user input (labeled _user_) or a [CSV marker][csv-markers] (labeled _marker_).
+This list may change as the SDK becomes better at generating CSV's.
 
 Required:
-
-* `metadata.name`: a *unique* name for this CSV. Operator version should be included in the name to ensure uniqueness, ex. `app-operator.v0.1.1`.
-* `spec.description` _(user)_ : a thorough description of the Operator's functionality.
-* `spec.displayName` _(user)_ : a name to display for the Operator in Operator Hub.
-* `spec.keywords` _(user)_ : a list of keywords describing the Operator.
-* `spec.maintainers` _(user)_ : a list of human or organizational entities maintaining the Operator, with a `name` and `email`.
-* `spec.provider` _(user)_ : the Operator provider, with a `name`; usually an organization.
-* `spec.labels` _(user)_ : a list of `key:value` pairs to be used by Operator internals.
-* `spec.version`: semantic version of the Operator, ex. `0.1.1`.
-* `spec.installModes`: what mode of [installation namespacing][install-modes] OLM should use. Currently all but `MultiNamespace` are supported by SDK Operators.
-* `spec.customresourcedefinitions`: any CRDs the Operator uses. Certain fields in elements of `owned` will be filled by the SDK.
-    * `owned`: all CRDs the Operator deploys itself from it's bundle.
-        * `name`: CRD's `metadata.name`.
-        * `kind`: CRD's `metadata.spec.names.kind`.
-        * `version`: CRD's `metadata.spec.version`.
-        * `description` _(marker)_ : description of the CRD.
-        * `displayName` _(marker)_ : display name of the CRD.
-        * `resources` _(marker)_ : any Kubernetes resources used by the CRD, ex. `Pod`'s and `ConfigMap`'s.
-        * `specDescriptors` _(marker)_ : UI hints for inputs and outputs of the Operator's spec.
-        * `statusDescriptors` _(marker)_ : UI hints for inputs and outputs of the Operator's status.
-        * `actionDescriptors` _(user)_ : UI hints for an Operator's in-cluster actions.
-    * `required` _(user)_ : all CRDs the Operator expects to be present in-cluster, if any. All `required` element fields must be populated manually.
+- `metadata.name`: a *unique* name for this CSV of the format `<operator-name>.vX.Y.Z`, ex. `app-operator.v0.0.1`.
+- `spec.version`: semantic version of the Operator, ex. `0.0.1`.
+- `spec.installModes`: what mode of [installation namespacing][install-modes] OLM should use.
+Currently all but `MultiNamespace` are supported by SDK Operators.
+- `spec.customresourcedefinitions`: any CRDs the Operator uses. Certain fields in elements of `owned` will be filled by the SDK.
+    - `owned`: all CRDs the Operator deploys itself from it's bundle.
+        - `name`: CRD's `metadata.name`.
+        - `kind`: CRD's `metadata.spec.names.kind`.
+        - `version`: CRD's `metadata.spec.version`.
```

Comment:
all the above three elements should be labelled _user_ ?
It smells to me that they are all required... I may be wrong though.

Reply:
Why wouldn't they be required? They define what CRDs are used by an operator. Plus these are always parsed from CRD manifests in the relevant directory.

Reply:
I mean ... if they are *required*, should we add the label `(_user_)` as we do for other elements?
Maybe it is too obvious, :).

Reply:
That tag is only added for fields the user has to write themselves. Name, version, and kind are all inferrable so a user doesn’t have to do anything. Required means they must be set for a CRD element or static validation will fail, which is unrelated to the former statement. Does that make sense?

Reply:
That makes perfect sense. Thanks for the patience and detailed explanation.